src/plots/cartesian/tick_label_defaults.js

```diff
@@ -43,6 +43,7 @@ module.exports = function handleTickLabelDefaults(containerIn, containerOut, coe
             if(!tickFormat && axType !== 'date') {
                 coerce('showexponent', showAttrDflt);
                 coerce('exponentformat');
+                coerce('separatethousands');
```

Comment:
`Lib.coerce` fails if a passed prop (here `separatethousands`) isn't part of the attribute object used to defined the `coerce` wrapper function above. 

So, we need to make sure that `separatethousands` is part of all attribute objects that pass through here. I believe this mean adding `seperatethousands` in:

- colorbar [attributes](https://github.com/plotly/plotly.js/blob/master/src/components/colorbar/attributes.js),
- gl3d axis [attributes](https://github.com/plotly/plotly.js/blob/master/src/plots/gl3d/layout/attributes.js), and
- ternary axis [attributes](https://github.com/plotly/plotly.js/blob/master/src/plots/ternary/layout/axis_attributes.js)

Moreover, to enable `seperatethousands` in color bars, you'll need to pass to the mock axis [here](https://github.com/plotly/plotly.js/blob/2fc55cfe734acfabc98955e6d3db7726d0e4e036/src/components/colorbar/draw.js#L143).